server/src/main/java/io/druid/segment/realtime/appenderator/BatchAppenderatorDriver.java

```diff
@@ -129,13 +129,11 @@ private SegmentsAndMetadata pushAndClear(
       long pushAndClearTimeoutMs
   ) throws InterruptedException, ExecutionException, TimeoutException
   {
-    final List<SegmentIdentifier> segmentIdentifierList = getSegmentWithStates(sequenceNames)
-        .filter(segmentWithState -> segmentWithState.getState() == SegmentState.APPENDING)
-        .map(SegmentWithState::getSegmentIdentifier)
-        .collect(Collectors.toList());
+    final Map<SegmentIdentifier, SegmentWithState> requestedSegmentIdsForSequences = getAppendingSegments(sequenceNames)
```

Comment:
What is the reason for moving the creation of `requestedSegmentIdsForSequences` from after the push, to before the push? Is it fixing something?